packages/material-ui/src/Modal/ModalManager.ts

```diff
@@ -84,54 +99,55 @@ function handleContainer(containerInfo, props) {
     const parent = container.parentElement;
     const containerWindow = ownerWindow(container);
     const scrollContainer =
-      parent.nodeName === 'HTML' &&
-      containerWindow.getComputedStyle(parent)['overflow-y'] === 'scroll'
+      parent?.nodeName === 'HTML' && containerWindow.getComputedStyle(parent).overflowY === 'scroll'
```

Comment:
It seems that we don't need this check after all: https://github.com/mui-org/material-ui/issues/22449#issuecomment-687057344

```suggestion
      parent.nodeName === 'HTML' && containerWindow.getComputedStyle(parent).overflowY === 'scroll'
```

Reply:
We do need it:

> The Node.parentElement read-only property returns the DOM node's parent Element, or null if the node either has no parent, or its parent isn't a DOM Element.

-- https://developer.mozilla.org/en-US/docs/Web/API/Node/parentElement

Reply:
What about letting it fail as an early feedback loop that it won't work later on with the styles? Maybe we could even add a warning in dev mode to explain that in a shadow DOM developer needs to nest deeper to host the styles?

Reply:
Sure, if this is a problem we can solve it in a follow-up. If `parent` is `null` then the subsequent checks wouldn't matter anyway which is why the guard is harmless.